cmd/cone/get_drop_task.go

```diff
@@ -49,17 +49,142 @@ func taskCmd(cmd *cobra.Command) *cobra.Command {
 	return cmd
 }
 
+func strToDur(duration string) (*time.Duration, error) {
+	if duration == "" {
+		return nil, nil
+	}
+
+	formattedDuration, err := str2duration.ParseDuration(duration)
+	if err != nil {
+		return nil, fmt.Errorf("invalid duration input %s", err.Error())
+	}
+
+	if formattedDuration <= time.Duration(0) {
+		return nil, errors.New("duration must be greater than 0")
+	}
+
+	return &formattedDuration, nil
+}
+
+func handleDurationNonInteractive(maxProvisionTime *time.Duration, duration string) (*time.Duration, error) {
+	formattedDuration, err := strToDur(duration)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := validateGrantTaskArguments(maxProvisionTime, formattedDuration); err != nil {
+		return nil, err
+	}
+
+	return formattedDuration, nil
+}
+
+func validateGrantTaskArguments(maxProvisionTime *time.Duration, duration *time.Duration) error {
+	// If maxProvisionTime is set, ensure the duration is not nil (which means infinite)
+	if maxProvisionTime != nil && duration == nil {
+		return fmt.Errorf("%s: %s", grantDurationErrorMessage, maxProvisionTime.String())
+	}
+
+	// If maxProvisionTime is set, ensure the duration is not greater than maxProvisionTime
+	if maxProvisionTime != nil && *duration > *maxProvisionTime {
+		return fmt.Errorf("%s: %s", grantDurationErrorMessage, maxProvisionTime.String())
+	}
+
+	return nil
+}
+
+func getValidDuration(ctx context.Context, v *viper.Viper, maxProvisionTime *time.Duration, duration string) (*time.Duration, error) {
+	// If both are empty that means they are both infinite
+	if maxProvisionTime == nil && duration == "" {
+		return nil, nil
+	}
+
+	if v.GetBool(nonInteractiveFlag) {
+		return handleDurationNonInteractive(maxProvisionTime, duration)
+	}
+	errorPrinter := pterm.PrefixPrinter{
+		MessageStyle: &pterm.ThemeDefault.ErrorMessageStyle,
+		Prefix: pterm.Prefix{
+			Style: &pterm.ThemeDefault.ErrorPrefixStyle,
+			Text:  " ERROR ",
+		},
+	}
+	input := pterm.DefaultInteractiveTextInput.WithMultiLine(false)
+	firstRun := true
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+			if !firstRun {
+				var err error
+				duration, err = input.Show()
+				if err != nil {
+					return nil, err
+				}
+			}
+			firstRun = false
+
+			formattedDuration, err := strToDur(duration)
+			if err != nil {
+				errorPrinter.Println(err.Error())
+				continue
+			}
+
+			err = validateGrantTaskArguments(maxProvisionTime, formattedDuration)
+			if err == nil {
+				return formattedDuration, nil
+			}
+			errorPrinter.Println(err.Error())
```

Comment:
I wasn't sure about this yesterday when we were writing this but I confirmed it now. You can do this:

```go
select {
case <-ctx.Done():
    return nil, ctx.Err()
default:
    break;
}

if !firstRun {

...
```

that way we remove an indentation